packages/core/src/html/scriptAndStyleTagProcessor.ts

```diff
@@ -10,24 +11,23 @@ const cheerio = require('cheerio');
  * @param node {Object<any, any>} node from the dom traversal
  * @param userScriptsAndStyles {array} to store scripts and style tags for hoisting
  */
-function processScriptAndStyleTag(node, userScriptsAndStyles) {
+export function processScriptAndStyleTag(node: DomElement, userScriptsAndStyles: string[]) {
   // Do not process script/style tags that are meant to be inserted in head/bottom of HTML
-  const isHeadOrBottom = node.parent.name === 'head-top'
-    || node.parent.name === 'head-bottom' || node.parent.name === 'script-bottom';
+  const isHeadOrBottom = node.parent && (node.parent.name === 'head-top'
+    || node.parent.name === 'head-bottom' || node.parent.name === 'script-bottom');
   // Do not process script/style tags that are from External
   const isExternal = userScriptsAndStyles === undefined;
   if (isHeadOrBottom || isExternal) {
     return;
   }
 
-  const idx = node.parent.children.indexOf(node);
-  if (idx !== -1) {
-    node.parent.children.splice(idx, 1);
+  const nodeSiblings = node.parent?.children;
```

Comment:
Personally I would do `nodeSiblings = node.parent ?? []` as it removes the need for the if check on the next line. Not a big deal, you can leave it if you want to.

Reply:
I like that better too, thanks for the suggestion!